libcst/_nodes/expression.py

```diff
@@ -1049,7 +1049,11 @@ def evaluated_value(self) -> Optional[str]:
         right_val = right.evaluated_value
         if right_val is None:
             return None
-        return left_val + right_val
+        if isinstance(left_val, bytes) and isinstance(right_val, bytes):
+            return left_val + right_val
+        if isinstance(left_val, str) and isinstance(right_val, str):
+            return left_val + right_val
+        return None
```

Comment:
Not entirely sure what to do with mixed values here, but I assume that's invalid Python and that returning `None` should be fine?